server/src/test/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java

```diff
@@ -782,12 +782,29 @@ public void testDisconnectsWhileRecovering() throws Exception {
             (MockTransportService) internalCluster().getInstance(TransportService.class, redNodeName);
         TransportService redTransportService = internalCluster().getInstance(TransportService.class, redNodeName);
         TransportService blueTransportService = internalCluster().getInstance(TransportService.class, blueNodeName);
-        final CountDownLatch requestBlocked = new CountDownLatch(1);
+        final CountDownLatch requestFailed = new CountDownLatch(1);
 
-        blueMockTransportService.addSendBehavior(redTransportService,
-            new RecoveryActionBlocker(dropRequests, recoveryActionToBlock, requestBlocked));
-        redMockTransportService.addSendBehavior(blueTransportService,
-            new RecoveryActionBlocker(dropRequests, recoveryActionToBlock, requestBlocked));
+        if (randomBoolean()) {
+            // Fail on the sending side
+            blueMockTransportService.addSendBehavior(redTransportService,
+                new RecoveryActionBlocker(dropRequests, recoveryActionToBlock, requestFailed));
+            redMockTransportService.addSendBehavior(blueTransportService,
+                new RecoveryActionBlocker(dropRequests, recoveryActionToBlock, requestFailed));
+        } else {
+            // Fail on the receiving side.
+            blueMockTransportService.addRequestHandlingBehavior(recoveryActionToBlock, (handler, request, channel, task) -> {
+                logger.info("--> preventing {} response", recoveryActionToBlock);
+                requestFailed.countDown();
+                blueMockTransportService.disconnectFromNode(redMockTransportService.getLocalDiscoNode());
```

Comment:
We would like the connection from red to blue to be disconnected, not the one from blue to red?